### DISCUSSION TITLE
Which access rights are needed to run prisma with postgres?

### DISCUSSION THREAD

**Discussion by dseemann** [prisma/prisma#2375] [Q&A] [answered]
We're trying to deploy prisma2 on a hosted postgres with Digital Ocean.

When trying to connect or migrate with a non-admin user, prisma aborts with `Error querying the database: db error: ERROR: permission denied for database DBNAME`

Connecting and migrating with the provided admin user is working fine.

Which access rights on a database are needed to work with non-admin users?

**Comment by ryands17** [comment, marked as answer]
Hey @5in4 :wave: 

The user created just requires the rights to the database that you will be using. So I think granting all privileges on the database should work fine as well.

**Reply by dseemann** [reply]
Hi!
Unfortunately, we're getting the error mentioned above. Using the same kind of user with other frameworks like e.g. django works fine.

Do you have a pointer on where to start debugging?

**Reply by janpio** [reply]
What commands or methods exactly are you trying to run?
Setting the env var `DEBUG=*` should give you a lot of useful output.

**Reply by pantharshit00** [reply]
@5in4 Also, please make sure you are actually able to query with that user in the first place.

**Reply by dseemann** [reply]
Hi!

It seems like granting all privileges on tables and sequences is not sufficient. We've got it running by running `GRANT ALL PRIVILEGES ON DATABASE dbname TO username;`.

Unfortunately, the error without these exact access rights is always `ERROR: permission denied for database`, so I'm not really sure what exactly was missing.

Anyways, thank you for your time!

**Comment by RyanHedges** [comment]
I had this error locally and [@5in4 comment](https://github.com/prisma/prisma/discussions/2375#discussioncomment-18457) helped. The error was the same when running the `dev` and `deploy` migration command on a new local db with limited schema and role permissions. 

```
Error: Migration engine error:
db error: ERROR: permission denied for database dbname
```

Executing `GRANT CREATE ON DATABASE dbname TO username;` worked for us if you don't want to provide all privileges. Also, just in case someone else comes across this later, for the `dev` command, we also executed `ALTER ROLE username WITH CREATEDB;` as our user was inheriting role permissions and `CREATEDB` isn't inherited [see psql docs](https://www.postgresql.org/docs/current/role-membership.html).

Our Prisma version:
```
prisma                  : 3.10.0
@prisma/client          : ^3.10.0
```

**Comment by MuktadirHassan** [comment]
The current docs does a good job explaining this error and providing a solution. Anyone searching for the solution now 
may follow the link. [Prisma Shadow Database](https://www.prisma.io/docs/concepts/components/prisma-migrate/shadow-database)
![image](https://user-images.githubusercontent.com/46109431/197931733-f1a67675-23d0-4909-b390-867fbb9b140b.png)

**Comment by ludralph** [comment]
Hi @5in4,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
